nixos/modules/system/boot/systemd.nix

```diff
@@ -219,7 +219,9 @@ let
         // optionalAttrs (config.documentation != []) {
           Documentation = toString config.documentation; }
         // optionalAttrs (config.onFailure != []) {
-          OnFailure = toString config.onFailure;
+          OnFailure = toString config.onFailure; }
+        // optionalAttrs (config.startLimitIntervalSec != []) {
```

Comment:
Remove the default, add an `options` argument at line 196, then change this here to `options.startLimitIntervalSec.isDefined`

Reply:
Thanks! It looks like the comparisons at https://github.com/NixOS/nixpkgs/blob/04b4ca37bd836aab790684c09319c05e7f3a84df/nixos/modules/system/boot/systemd.nix#L199 are against their defaults. Should they be changed to use isDefined?

Reply:
I'm actually not sure now what would be better

| Property | Comparing with default | Using `isDefined` | Importance (subjective) |
| --- | --- | --- | --- |
| Defining as default doesn't trigger generation | Yes | No | 2 |
| Is usually done | Yes | No | 5 |
| Can use value without having set it (getting an empty list makes sense and is useful) | Yes | No | 10 |
| No need to duplicate upstream defaults | No | Yes | 10 |
| Reflects the fact that set values have an influence, unset ones don't | No | Yes | 3 |

Both have advantages and disadvantages. Assigning some subjective importances to these properties, it's 17 vs. 13. The fact that we don't need to duplicate upstream defaults is a pretty good seller for me, but this also means we can't use these config values without having set them, which can be very useful.

After this consideration, I think it would be better to compare everything with the default value instead, so you could change it back again to the original way and use the upstream default is the options default. Now maybe other people have different opinions, but this is mine :)